qlib/workflow/exp.py

```diff
@@ -197,14 +210,67 @@ def delete_recorder(self, recorder_id=None, recorder_name=None):
                 self.client.delete_run(recorder_id)
             else:
                 recorders = self.list_recorders()
-                for r in recorders:
-                    if recorders[r].name == recorder_name:
-                        recorder_id = r
-                        break
-                self.client.delete_run(recorder_id)
-        except:
+                recorder = self._get_recorder_by_name(recorder_name)
+                self.client.delete_run(recorder.id)
+        except MlflowException as e:
+            raise Exception(
+                f"Error: {e}. Something went wrong when deleting recorder. Please check if the name/id of the recorder is correct."
+            )
+
+    def _get_recorder_by_id(self, recorder_id=None, create=False):
+        """
+        Get a recorder by its id. If the `create` is set to True, this method will also start to run the recorder.
+
+        Parameters
+        ----------
+        recorder_id : str
+            the id of the recorder to be returned.
+        create : boolean
+            create the recorder if it hasn't been created before.
+
+        Returns
+        -------
+        The specific recorder with given id.
+        """
+        recorders = self.list_recorders()
+        if recorder_id in recorders:
+            return recorders[recorder_id]
+        else:
+            if create:
+                logger.warning(f"No valid recorder found. Create a new recorder with name {recorder_name}.")
+                self.start(recorder_name)
+                return self.active_recorder
+            else:
+                raise Exception(
+                    "Something went wrong when retrieving recorders. Please check if id of the recorder is correct."
+                )
+
+    def _get_recorder_by_name(self, recorder_name=None, create=False):

```

Comment:
I think things could be simpler if we use a function named _get_recorder_by_or_name